### ISSUE TITLE
[feature-request] Blue Ocean UI integration

### ISSUE THREAD

**Issue by etki** [jenkinsci/allure-plugin#68]
I know that's long long way to go and will spend quite some time in backlog in best case, but it would be terribly nice to have blue ocean UI tab for allure reports.

**Comment by nofate** [comment]
Would love this too.

**Comment by denis-platonov** [comment]
Agree, I was searching for this feature and found this request, so +1 here

**Comment by papitoio** [comment]
me too

**Comment by pumano** [comment]
@eroshenkoam any news about this feature request?

**Comment by Noksa** [comment]
@eroshenkoam 
Up

**Comment by efpato** [comment]
+1

**Comment by rickschubert** [comment]
👍

**Comment by MikhailGordeev** [comment]
look forward to 👍

**Comment by elzenar** [comment]
+1

**Comment by dimmel82** [comment]
+1

**Comment by cntasios** [comment]
+1

**Comment by thiagotrentin** [comment]
+1

**Comment by etki** [comment]
Just posting my findings here. As for now, this looks pretty much like impossible task. Blue Ocean UI exposes extension points, which plugins may use to hook unto, obviously for one to appear it should be manually declared in code. Blue Ocean UI does not expose extension points neither for top bar

![image](https://user-images.githubusercontent.com/1909762/71497015-4dc94500-2867-11ea-9dc8-03759f6d2430.png)
https://github.com/jenkinsci/blueocean-plugin/blob/blueocean-parent-1.21.0/blueocean-dashboard/src/main/js/components/RunDetails.jsx#L172

or anywhere in the artifact list (so there is no option to place additional table with reports only)

![image](https://user-images.githubusercontent.com/1909762/71497074-a13b9300-2867-11ea-8f98-b5f3145f416e.png)
https://github.com/jenkinsci/blueocean-plugin/blob/blueocean-parent-1.21.0/blueocean-dashboard/src/main/js/components/RunDetailsArtifacts.jsx

The only thing to hook up is adding new routes using `jenkins.main.routes` extension point. But the problem is, Jenkins uses nested routes to display changes, artifacts and etc. in tabs (i.e. the outer view does not change, only inner component). There is no way to inject anything [there](https://github.com/jenkinsci/blueocean-plugin/blob/blueocean-parent-1.21.0/blueocean-dashboard/src/main/js/PipelineRoutes.jsx#L141), so while it should be possible to build custom page that fetches artifact content, there is no way to add a link for that on pipeline page. Also i'm not sure it is easy to call modules from plugin A (pipeline dashboard) that belong to plugin B (artifact viewer in that case).

Issues and/or PRs to Blue Ocean repo that will add necessary extension points are appreciated (i am afraid that i has exceeded my time limit for digging this).

**Comment by etki** [comment]
On the other though https://issues.jenkins-ci.org/browse/JENKINS-60592
